src/contracts/dependencies/StandardToken.ts

```diff
@@ -53,7 +57,11 @@ export class StandardToken extends Contract {
    * @param amount The amount to transfer
    */
   public transfer(from: Address, to: Address, amount: BigNumber) {
-    return this.createTransaction('transfer', from, [to, amount.toString()]);
+    return this.createTransaction('transfer', from, [to, amount.toString()], undefined, async () => {
+      const balance = await this.getBalanceOf(from);
+      validate(amount.isLessThanOrEqualTo(balance), new OutOfBalanceError(balance.toNumber()));
```

Comment:
I am not sure if I like this `validate` utility. Usually this pattern is used as `invariant` and that is only used in most cases to allow for different handling of errors in different environments and for minification of the bundle. In our case, it just makes the readability of the code worse. A simple `if (foo) { ... }` condition with a `throw new Bar()` inside would be preferable imho.

Reply:
Just wanted to mimic the 'require' in solidity. But you're right it would be harder for javascript devs to read the code at first

Reply:
Also how would one go about just checking that all the require for a specific transaction return true (before sending  any tx)

Reply:
> For transactions that have +4 require, would you still go for the if statements instead of this `validate` utility @fubhy

Yes :-)

> Also how would one go about just checking that all the require for a specific transaction return true (before sending any tx)

By calling `await transaction.validate()` before running `await transaction.send()`